controllers/util/tests/tests.go

```diff
@@ -0,0 +1,171 @@
+
+	containersImages = addImagesByNameFromYaml(containersImages, c.getCrYaml())
+	return containersImages
+}
+
+func (c *crYamlConfig) getCrYaml() *crYamlConfig {
+	yamlFile, err := ioutil.ReadFile(relativeCrPath)
+	if err != nil {
+		fmt.Println("unable to read yaml file, error:", err)
+	 	os.Exit(1)
+	}
+	err = yaml.Unmarshal(yamlFile, c)
+	if err != nil {
+		fmt.Println("unable to decodes yaml file, error:", err)
+	 	os.Exit(1)
+	}
+	 return c
+}
+ 
+func addImagesByNameFromYaml(containersImages map[string]string, crYaml *crYamlConfig) map[string]string {
+	containersImages = addSideCarsImagesToContainersImagesMap(containersImages, crYaml)
+	containersImages = addNodeImageToContainersImagesMap(containersImages, crYaml)
+	containersImages = addControllerImageToContainersImagesMap(containersImages, crYaml)
+	return containersImages
+}
+ 
+func addSideCarsImagesToContainersImagesMap(containersImages map[string]string, crYaml *crYamlConfig) map[string]string {
+	for _, sidecar := range crYaml.Spec.Sidecars {
+		containersImages[sidecar.Name] = getImageWithoutForwardSlash(sidecar)
+	}
+	return containersImages
+}
+
+func addNodeImageToContainersImagesMap(containersImages map[string]string, crYaml *crYamlConfig) map[string]string {
+	node := crYaml.Spec.Node
+	containersImages[nodeContainerName] = getImageWithoutForwardSlash(node)
+	return containersImages
+}
+
+func addControllerImageToContainersImagesMap(containersImages map[string]string, crYaml *crYamlConfig) map[string]string {
+	controller := crYaml.Spec.Controller
+	containersImages[controllerContainerName] = getImageWithoutForwardSlash(controller)
+	return containersImages
+}
+
+func getImageWithoutForwardSlash(container imageProperties) string {
+	image := container.Repository + ":" + container.Tag
+	return strings.Replace(image, "/", "-", -1)
+}
+
+func GetIbmBlockCsiSpec(containersImages map[string]string) csiv1.IBMBlockCSISpec {
+	var spec csiv1.IBMBlockCSISpec
+	spec.Controller = addControllerToIbmSpec(spec, containersImages)
+	spec.Node = addNodeToIbmSpec(spec, containersImages)
+	spec.Sidecars = addSidecarsToIbmSpec(spec, containersImages)
+	return spec
+}
+
+func addControllerToIbmSpec(spec csiv1.IBMBlockCSISpec, containersImages map[string]string) csiv1.IBMBlockCSIControllerSpec {
+	var controllerSpec csiv1.IBMBlockCSIControllerSpec
+	controllerSpec.Repository = strings.Split(containersImages[controllerContainerName], ":")[0]
+	controllerSpec.Tag = strings.Split(containersImages[controllerContainerName], ":")[1]
+	return controllerSpec
+}
+
+func addNodeToIbmSpec(spec csiv1.IBMBlockCSISpec, containersImages map[string]string) csiv1.IBMBlockCSINodeSpec {
+	var nodeSpec csiv1.IBMBlockCSINodeSpec
+	nodeSpec.Repository = strings.Split(containersImages[nodeContainerName], ":")[0]
+	nodeSpec.Tag = strings.Split(containersImages[nodeContainerName], ":")[1]
+	return nodeSpec
+}
+
+func addSidecarsToIbmSpec(spec csiv1.IBMBlockCSISpec, containersImages map[string]string) []csiv1.CSISidecar {
```

Comment:
```suggestion
func addSidecarsToIBMSpec(spec csiv1.IBMBlockCSISpec, containersImages map[string]string) []csiv1.CSISidecar {
```
(same for all `Ibm`)

Reply:
same for all `Csi`

Reply:
done